### PR TITLE
Mock CompetitionInfo and create more Test Competitions

### PR DESCRIPTION
**Pull request by FinnIckler**
Creates some Test Competitions that have edge cases on them. 
I also learned that there are a lot of fields that can be null, something like using_stripe_payments? for example is not true or false, it's true or null....
While adding the favourites test competition I noticed a bug where main_event_id is not set (also did not know this was possible) so I fixed it.

### CODE REVIEW

File: Frontend/src/api/mocks/fixtures.ts
Comment (kr-matthews) [inline Frontend/src/api/mocks/fixtures.ts:1]: Won't these get out-of-date? Ie `NOT_YET_OPEN` has a concrete open date, which will eventually be in the past.

File: Frontend/src/api/mocks/get_competition_info.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      return NOT_YET_OPEN
+    }
+    case 'FMCFrance2023': {
+      return COMMENT_REQUIRED
+    }
+    case 'PickeringFavouritesAutumn2023': {
+      return FAVOURITES_COMPETITION
+    }
+    default: {
+      // This allows non mocked response when debugging a certain competition
```
Comment (kr-matthews) [inline Frontend/src/api/mocks/get_competition_info.ts:35]: The function name is a bit misleading then.

File: Frontend/src/api/types.d.ts [hunk 2]
```diff
@@ -72,21 +74,24 @@ interface CompetitionEvent {
...
-  email: string
-  region: string
-  senior_delegate_id: number
+  location?: string
+  email?: string
+  region?: string
+  senior_delegate_id?: number
   class: string
+  url: string
   //TODO
```
Comment (kr-matthews) [inline Frontend/src/api/types.d.ts:90]: Is this still relevant?

File: Frontend/src/ui/Competition.jsx [hunk 3]
```diff
@@ -129,15 +129,17 @@ export default function Competition({ children }) {
...
-                </span>
-              </div>
+              {competitionInfo.main_event_id && (
+                <div>
+                  <span className={styles.eventHeader}>Main Event:</span>
+                  <span className={styles.event}>
+                    <CubingIcon
+                      event={competitionInfo.main_event_id}
+                      selected={true}
+                    />
```
Comment (kr-matthews) [inline Frontend/src/ui/Competition.jsx:139]: We really need a 'non-selectable' version of this.
```suggestion
                    <CubingIcon
                      event={competitionInfo.main_event_id}
                      selected
                    />
```

Comment (FinnIckler) [reply Frontend/src/ui/Competition.jsx:139]: It isn't actually selectable by default in my code. That the monolith overwriting my CSS

File: Frontend/src/api/types.d.ts (hunk 2 above)
Comment (FinnIckler) [reply Frontend/src/api/types.d.ts:90]: added actual type

File: Frontend/src/api/mocks/fixtures.ts
Comment (FinnIckler) [reply Frontend/src/api/mocks/fixtures.ts:1]: used moment to make sure that won't happen

File: Frontend/src/api/mocks/get_competition_info.ts (hunk 1 above)
Comment (FinnIckler) [reply Frontend/src/api/mocks/get_competition_info.ts:35]: I would like to keep that functionality though, so we can more easily handle bug reports. Do you have any good idea for a name?

Comment (kr-matthews) [reply Frontend/src/api/mocks/get_competition_info.ts:35]: Maybe add `...WithRealFallback` on to the end, or something to that effect?

Comment (FinnIckler) [reply Frontend/src/api/mocks/get_competition_info.ts:35]: renamed
